src/Refinement/RefinedTriangulations.jl

```diff
@@ -0,0 +1,126 @@
+
+
+struct RefinedTriangulation{Dc,Dp,A<:Triangulation{Dc,Dp},B<:RefinedDiscreteModel} <: Triangulation{Dc,Dp}
```

Comment:
I agree that here we need the flexibility of having a `RefinedDiscreteModel` with `Dc` different from the one of `Triangulation`.  In any case, there are still some preconditions of `RefinedDiscreteModel` that have to be fullfilled against `Triangulation`. In particular, they both have to have the same `Dp`, and `Dc_triangulation` <= `Dc_model`. Are we willing to explicitly control that thes preconditions are fullfilled? (e.g., in a explicit inner constructor).

Reply:
Resolved by [9865259](https://github.com/gridap/Gridap.jl/pull/838/commits/98652592b99cbdf691c38066c15dc47e1f8d5d71)